tests/unittests/OperatorTest.cpp

```diff
@@ -2494,6 +2494,238 @@ TEST_P(OperatorTest, ScatterAssignQuantized) {
+  //                          [-10., -11., -12.]];
+  //
+  // 2. By index [1, 1], set [[-13., -14., -15.]
+  //                          [-16., -17., -18.]
+  //                          [-19., -20., -21.]
+  //                          [-22., -23., -24.]];
+  //
+  auto *data =
+      mod_.createPlaceholder(ElemKind::FloatTy, {2, 4, 4, 3}, "data", false);
+  auto *indices =
+      mod_.createPlaceholder(ElemKind::Int64ITy, {2, 2}, "indices", false);
+  auto *slices =
+      mod_.createPlaceholder(ElemKind::FloatTy, {2, 4, 3}, "slices", false);
+
+  // Fill tensor with consecutive data.
+  std::vector<float> init(2 * 4 * 4 * 3);
+  std::iota(init.begin(), init.end(), 0);
+  bindings_.allocate(data)->getHandle() = init;
+  bindings_.allocate(indices)->getHandle<int64_t>() = {0, 3, 1, 1};
+  std::vector<float> initUpdates;
+  for (int32_t i = -1; i > -25; i--) {
+    initUpdates.push_back(static_cast<float>(i));
+  }
+  bindings_.allocate(slices)->getHandle() = initUpdates;
+
+  auto *R = F_->createScatterData("scatterdata", data, indices, slices);
+
+  auto *result = F_->createSave("save", R);
+  bindings_.allocate(result->getPlaceholder());
+
+  EE_.compile(CompilationMode::Infer, F_);
+  EE_.run(bindings_);
+
+  std::vector<size_t> expectedDims = {2, 4, 4, 3};
+  std::vector<float> expectedValues = {
+      0.0f,   1.0f,   2.0f,   3.0f,   4.0f,   5.0f,
+      6.0f,   7.0f,   8.0f,   9.0f,   10.0f,  11.0f,
+
+      12.0f,  13.0f,  14.0f,  15.0f,  16.0f,  17.0f,
+      18.0f,  19.0f,  20.0f,  21.0f,  22.0f,  23.0f,
+
+      24.0f,  25.0f,  26.0f,  27.0f,  28.0f,  29.0f,
+      30.0f,  31.0f,  32.0f,  33.0f,  34.0f,  35.0f,
+
+      -1.0f,  -2.0f,  -3.0f,  -4.0f,  -5.0f,  -6.0f,
+      -7.0f,  -8.0f,  -9.0f,  -10.0f, -11.0f, -12.0f,
+
+      48.0f,  49.0f,  50.0f,  51.0f,  52.0f,  53.0f,
+      54.0f,  55.0f,  56.0f,  57.0f,  58.0f,  59.0f,
+
+      -13.0f, -14.0f, -15.0f, -16.0f, -17.0f, -18.0f,
+      -19.0f, -20.0f, -21.0f, -22.0f, -23.0f, -24.0f,
+
+      72.0f,  73.0f,  74.0f,  75.0f,  76.0f,  77.0f,
+      78.0f,  79.0f,  80.0f,  81.0f,  82.0f,  83.0f,
+
+      84.0f,  85.0f,  86.0f,  87.0f,  88.0f,  89.0f,
+      90.0f,  91.0f,  92.0f,  93.0f,  94.0f,  95.0f};
+  auto H = bindings_.get(result->getPlaceholder())->getHandle();
+  EXPECT_TRUE(H.dims().vec() == expectedDims);
+  for (size_t i = 0; i < expectedValues.size(); i++) {
+    EXPECT_EQ(expectedValues[i], H.raw(i));
+  }
+}
+
+TEST_P(OperatorTest, ScatterAddQuantized) {
+  ENABLED_BACKENDS(Interpreter, CPU);
+
+  auto *data = mod_.createPlaceholder(ElemKind::FloatTy, {5, 2}, "data", false);
+  auto *indices =
+      mod_.createPlaceholder(ElemKind::Int64ITy, {2, 1}, "indices", false);
+  auto *slices =
+      mod_.createPlaceholder(ElemKind::FloatTy, {2, 2}, "slices", false);
+
+  bindings_.allocate(data)->getHandle() = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+  bindings_.allocate(indices)->getHandle<int64_t>() = {1, 3};
+  bindings_.allocate(slices)->getHandle() = {-3, -4, -7, -8};
```

Comment:
Can you make `-8` positive so we verify that the kernels correctly handle maxing out the quantized range?

Reply:
Hm, looks like they don't. Good catch! I'll fix this.